### PR TITLE
Upgrade to Alpha Staking V3 contract, unit tests and deploy script

### PR DESCRIPTION
**Pull request by HermioneETH**

### CODE REVIEW

File: contracts/AlphaStakingV3.sol [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      data.status = STATUS_READY;
+      data.unbondTime = 0;
+      data.unbondShare = 0;
+    }
+    alpha.safeTransferFrom(msg.sender, address(this), amount);
+    share = totalAlpha == 0 ? amount : amount.mul(totalShare).div(totalAlpha);
+    totalAlpha = totalAlpha.add(amount);
+    totalShare = totalShare.add(share);
+    data.share = data.share.add(share);
+    emit Stake(owner, share, amount);
```
Comment (n1punp) [inline contracts/AlphaStakingV3.sol:106]: Let's separate normal Stake and StakerFor events, so we can separately track staking from claims in Staking contract.

File: contracts/AlphaStakingV3.sol [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  function getStakeValue(address user) external view returns (uint) {
+    uint share = users[user].share;
+    return share == 0 ? 0 : share.mul(totalAlpha).div(totalShare);
+  }
+
+  function stake(uint amount) external nonReentrant {
+    _stake(msg.sender, amount);
+  }
+
+  function _stake(address owner, uint amount) internal returns (uint share) {
```
Comment (n1punp) [inline contracts/AlphaStakingV3.sol:92]: No need to return shares

File: contracts/AlphaStakingV3.sol (hunk 1 above)
Comment (HermioneETH) [reply contracts/AlphaStakingV3.sol:107]: Fixed

File: contracts/AlphaStakingV3.sol (hunk 2 above)
Comment (HermioneETH) [reply contracts/AlphaStakingV3.sol:93]: Fixed

File: contracts/AlphaStakingV3.sol [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    emit SetMerkle(_merkle);
+  }
+
+  function acceptGovernor() external {
+    require(msg.sender == pendingGovernor, 'acceptGovernor/not-pending');
+    pendingGovernor = address(0);
+    governor = msg.sender;
+  }
+
+  function getStakeValue(address user) external view returns (uint) {
```
Comment (nkitlabs) [inline contracts/AlphaStakingV3.sol:84]: change input parameter to have underscore at first i.e. `function getStakeValue(address _user)`

File: interfaces/IAny.sol [hunk 4]
```diff
@@ -59,4 +59,8 @@ interface IAny {
     function extract(uint) external;
 
     function getProxyImplementation(address) external view returns (address);
+
+    function setMerkle(address) external;
+
+    function stakeFor(address owner, uint amount) external returns (uint share);
```
Comment (nkitlabs) [inline interfaces/IAny.sol:65]: stakeFor return nothing.

**Review by nkitlabs — APPROVED** [review]

**Pull request merged**